### ISSUE TITLE
Auto-generate publication list using ORCID & DOIs

### ISSUE THREAD

**Issue by DanNBullock** [academicpages/academicpages.github.io#1093]
Chris Holdgraf (not tagged here, due to likely already being spammed via GitHub interfaces) recently made [a blog post about using ORCID API tools and doi.org](https://chrisholdgraf.com/blog/2022/orcid-auto-update/) to auto-generate publication lists using ORCIDs.  I believe he is using python to achieve this, but perhaps there are other approaches to achieving this same outcome.

There is [an issue thread from 2017](https://github.com/academicpages/academicpages.github.io/issues/2) (indeed, issue #2), that seems somewhat stale, with minimal discussion / updates on the main branch (though there are various references on personal forks).

One potential challenge for this is that [the ORCID API appears to require registration / credential authentication](https://info.orcid.org/documentation/features/public-api/), which may prevent this approach from scaling well / easily.

**Comment by rjzupkoii** [comment]
Not sure if this can be done, but it's at least something we can take a closer look at.

**Comment by c-martinez** [comment]
I'm pretty sure it is possible to retrieve citations from orcid API, would it be useful if I write a short script for this and submit a PR?

**Comment by rjzupkoii** [comment]
@c-martinez Yes, that would be very helpful and I'm always open to PRs that improve the template. Something to keep in mind is that depending on when this is done as part of the workflow, you may be running in a container which can put some pretty significant limits on what tools / libraries are available and can be used. However, if you are looking at it from the perspective of generating all of the `*.md` files in `_publications` that might be much simpler and I would suggest putting the script that you use for that in a `scripts` directory so that we can make sure that it doesn't get deployed as part of a users website.

**Comment by c-martinez** [comment]
Great, good to know!

Fetching data from orcid API turned out to be fairly straight forward and we can generate a `*.bib` file with all the orcid entries. I'm guess `*.md` files could be generated by the existing scripts.

**Comment by RCPhysics** [comment]
This is likely an issue on my end, as I'm not a frequent python user.  When trying to use the notebook to generate the bibfile, I get this error at input 3:
---------------------------------------------------------------------------
NameError                                 Traceback (most recent call last)
Cell In[3], line 1
----> 1 response = requests.get('https://pub.orcid.org/v3.0/{}/works'.format(orcid),
      2                         headers={"Accept": "application/orcid+json" })
      3 record = response.json()

NameError: name 'orcid' is not defined

Am I supposed to define this somewhere else?

**Comment by rjzupkoii** [comment]
@RCPhysics You might want to ask again in the [discussions](https://github.com/academicpages/academicpages.github.io/discussions) to get a bit more people to look at it, but are you setting the `orcid` in the first cell of the notebook and then running each cell?

**Comment by RCPhysics** [comment]
@rjzupkoii Okay, yea, it was an error on my part.  I thought the first cell was running, but it was not.
